internal/controller/postgrescluster/postgres_test.go

```diff
@@ -251,26 +252,187 @@ func TestReconcilePostgresVolumes(t *testing.T) {
 		Owner:  client.FieldOwner(t.Name()),
 	}
 
-	cluster := testCluster()
-	cluster.Namespace = setupNamespace(t, tClient).Name
+	t.Run("DataVolumeNoSourceCluster", func(t *testing.T) {
+		cluster := testCluster()
+		ns := setupNamespace(t, tClient)
+		cluster.Namespace = ns.Name
 
-	assert.NilError(t, tClient.Create(ctx, cluster))
-	t.Cleanup(func() { assert.Check(t, tClient.Delete(ctx, cluster)) })
+		assert.NilError(t, tClient.Create(ctx, cluster))
+		t.Cleanup(func() { assert.Check(t, tClient.Delete(ctx, cluster)) })
 
-	spec := &v1beta1.PostgresInstanceSetSpec{}
-	assert.NilError(t, yaml.Unmarshal([]byte(`{
-		name: "some-instance",
-		dataVolumeClaimSpec: {
-			accessModes: [ReadWriteOnce],
-			resources: { requests: { storage: 1Gi } },
-			storageClassName: "storage-class-for-data",
-		},
-	}`), spec))
+		spec := &v1beta1.PostgresInstanceSetSpec{}
+		assert.NilError(t, yaml.Unmarshal([]byte(`{
+			name: "some-instance",
+			dataVolumeClaimSpec: {
+				accessModes: [ReadWriteOnce],
+				resources: { requests: { storage: 1Gi } },
+				storageClassName: "storage-class-for-data",
+			},
+		}`), spec))
+		instance := &appsv1.StatefulSet{ObjectMeta: naming.GenerateInstance(cluster, spec)}
+
+		pvc, err := reconciler.reconcilePostgresDataVolume(ctx, cluster, spec, instance, nil, nil)
+		assert.NilError(t, err)
+
+		assert.Assert(t, metav1.IsControlledBy(pvc, cluster))
+
+		assert.Equal(t, pvc.Labels[naming.LabelCluster], cluster.Name)
+		assert.Equal(t, pvc.Labels[naming.LabelInstance], instance.Name)
+		assert.Equal(t, pvc.Labels[naming.LabelInstanceSet], spec.Name)
+		assert.Equal(t, pvc.Labels[naming.LabelRole], "pgdata")
+
+		assert.Assert(t, marshalMatches(pvc.Spec, `
```

Comment:
🏕️ Do you mind changing the ones in this test to [cmp.MarshalMatches](https://github.com/CrunchyData/postgres-operator/blob/b4eb42aa3af6ef917755183bb78c4442bd33245f/internal/testing/cmp/cmp.go#L65) while you're here?

```suggestion
		assert.Assert(t, cmp.MarshalMatches(pvc.Spec, `
```